docs/design.md

```diff
@@ -414,27 +444,115 @@ Procedure:
 
 2. Inspect image trailers; is a swap requested?
     Yes.
-        1. Is the requested image valid (integrity and security check)?
-            Yes.
-                a. Perform swap operation.
-                b. Persist completion of swap procedure to image trailers.
-                c. Proceed to step 3.
-            No.
-                a. Erase invalid image.
-                b. Persist failure of swap procedure to image trailers.
-                c. Proceed to step 3.
+    ​    1. Is the requested image valid (integrity and security check)?
+    ​        Yes.
+    ​            a. Perform swap operation.
+    ​            b. Persist completion of swap procedure to image trailers.
+    ​            c. Proceed to step 3.
+    ​        No.
+    ​            a. Erase invalid image.
+    ​            b. Persist failure of swap procedure to image trailers.
+    ​            c. Proceed to step 3.
     No: Proceed to step 3.
 
-3. Boot into image in slot 0.
+3. Boot into image in primary slot.
+
+### Multiple Image Boot
+
+When the flash contains multiple executable images the boot loader's operation
+is a bit more complex but similar to the previously described procedure with
+one image. Every image can be updated independently therefore the flash is
+partitioned further to arrange two slots for each image.
+```
++--------------------+
+| MCUBoot            |
++--------------------+
+        ~~~~~            <- memory might be not contiguous
++--------------------+
+| Image 0            |
```

Comment:
How much work would it be to allow the primary and secondary slots of an image to be placed apart from one another, e.g. placing all the primary slots together, and all the secondary slots together? I think it will be easier for build systems if the active images don't have to be interspersed with secondary slots.

Reply:
This can be the done by configuring the build system in the way how you suggested. MCUboot gets the address of the  slots from the platform layer.  So it is up to the user. In the design doc it is just a possible memory layout.

Reply:
I would say minimal. Nothing depends on the exact location of the image slots in the flash. Every time it executes flash operations it uses the addresses and sizes from the flash map based on the flash area id.

Reply:
Ok, thanks @tamasban and @davidvincze. Good to hear it's flexible.